### ISSUE TITLE
Error at Launch of VS Code with extension enabled: The connection to terminal's pty host process is unresponsive the terminals may stop working

### ISSUE THREAD

**Issue by the-joey-o** [PowerShell/vscode-powershell#4581]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [ ] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [ ] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [ ] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

I have a brand new computer, new install of windows, and only one extension that's completely unconfigured. Why am I getting this error? 
`The connection to terminal's pty host process is unresponsive the terminals may stop working`
![image](https://github.com/PowerShell/vscode-powershell/assets/7991004/af0fd190-8b84-4e21-acaf-7f5657e1fb02)

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      5.1.19041.2673
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.2673
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Visual Studio Code Version

```console
1.78.2
x64
```

### Extension Version

```console
ms-vscode.powershell@2023.5.0
```

### Steps to Reproduce

Launch VSCode. Should occur at startup.

### Logs

5/17/2023 3:45:36 PM [NORMAL] - Visual Studio Code v1.78.2 64-bit

5/17/2023 3:45:36 PM [NORMAL] - Creating log directory at: 'file:///c%3A/Users/jodell/AppData/Roaming/Code/User/globalStorage/ms-vscode.powershell/logs/1684356336-8646b9dc-8b56-4dc7-99b9-6c4f8ee4d03f1684356335339'

5/17/2023 3:45:36 PM [NORMAL] - PowerShell Extension v2023.5.0

5/17/2023 3:45:36 PM [NORMAL] - Operating System: Windows 64-bit

5/17/2023 3:45:37 PM [NORMAL] - Language server starting --

5/17/2023 3:45:37 PM [NORMAL] -     PowerShell executable: C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe

5/17/2023 3:45:37 PM [NORMAL] -     PowerShell args: -NoProfile -ExecutionPolicy Bypass -Command Import-Module 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2023.5.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules' -EnableConsoleRepl -StartupBanner "PowerShell Extension v2023.5.0
Copyright (c) Microsoft Corporation.

https://aka.ms/vscode-powershell
Type 'help' to get help.
" -LogLevel 'Normal' -LogPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\logs\1684356336-8646b9dc-8b56-4dc7-99b9-6c4f8ee4d03f1684356335339\EditorServices.log' -SessionDetailsPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\sessions\PSES-VSCode-8496-592897.json' -FeatureFlags @() 

5/17/2023 3:45:37 PM [NORMAL] -     PowerShell Editor Services args: Import-Module 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2023.5.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules' -EnableConsoleRepl -StartupBanner "PowerShell Extension v2023.5.0
Copyright (c) Microsoft Corporation.

https://aka.ms/vscode-powershell
Type 'help' to get help.
" -LogLevel 'Normal' -LogPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\logs\1684356336-8646b9dc-8b56-4dc7-99b9-6c4f8ee4d03f1684356335339\EditorServices.log' -SessionDetailsPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\sessions\PSES-VSCode-8496-592897.json' -FeatureFlags @() 

5/17/2023 3:45:37 PM [NORMAL] - powershell.exe started.

5/17/2023 3:47:24 PM [NORMAL] - powershell.exe PID: 15272

5/17/2023 3:47:24 PM [NORMAL] - Waiting for session file...

5/17/2023 3:47:54 PM [WARNING] - Loading the PowerShell extension is taking longer than expected. If you're using privilege enforcement software, this can affect start up performance.

5/17/2023 3:51:25 PM [NORMAL] - Timed out waiting for session file to appear!

5/17/2023 3:51:29 PM [ERROR] - PowerShell language server process didn't start!

**Comment by andschwa** [comment]
Seems like you have a version that should work.

Could you please provide more [logging and environment information by following these steps](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment)?

Specifically I'm looking for any logs produced by PSES (the server / PowerShell process) to see how far along it got starting up.

Thanks!

**Comment by the-joey-o** [comment]
5/18/2023 8:31:53 AM [DIAGNOSTIC] - Global storage URI: 'vscode-userdata:/c%3A/Users/jodell/AppData/Roaming/Code/User/globalStorage/ms-vscode.powershell', log file path: 'file:///c%3A/Users/jodell/AppData/Roaming/Code/User/globalStorage/ms-vscode.powershell/logs/1684416713-a9fc9243-336d-40f5-a653-a94da5d0a2011684416708327/vscode-powershell.log'

5/18/2023 8:31:53 AM [DIAGNOSTIC] - Loaded settings:
{
  "powerShellAdditionalExePaths": {},
  "powerShellDefaultVersion": "",
  "powerShellExePath": "",
  "promptToUpdatePowerShell": true,
  "startAsLoginShell": {
    "osx": true,
    "linux": false
  },
  "startAutomatically": true,
  "enableProfileLoading": true,
  "helpCompletion": "BlockComment",
  "scriptAnalysis": {
    "enable": true,
    "settingsPath": "PSScriptAnalyzerSettings.psd1"
  },
  "debugging": {
    "createTemporaryIntegratedConsole": false
  },
  "developer": {
    "featureFlags": [],
    "bundledModulesPath": "../../PowerShellEditorServices/module",
    "editorServicesLogLevel": "Diagnostic",
    "editorServicesWaitForDebugger": false,
    "waitForSessionFileTimeoutSeconds": 240
  },
  "codeFormatting": {
    "autoCorrectAliases": false,
    "avoidSemicolonsAsLineTerminators": false,
    "preset": "Custom",
    "openBraceOnSameLine": true,
    "newLineAfterOpenBrace": true,
    "newLineAfterCloseBrace": true,
    "pipelineIndentationStyle": "NoIndentation",
    "whitespaceBeforeOpenBrace": true,
    "whitespaceBeforeOpenParen": true,
    "whitespaceAroundOperator": true,
    "whitespaceAfterSeparator": true,
    "whitespaceBetweenParameters": false,
    "whitespaceInsideBrace": true,
    "addWhitespaceAroundPipe": true,
    "trimWhitespaceAroundPipe": false,
    "ignoreOneLineBlock": true,
    "alignPropertyValuePairs": true,
    "useConstantStrings": false,
    "useCorrectCasing": false
  },
  "integratedConsole": {
    "showOnStartup": true,
    "startInBackground": false,
    "focusConsoleOnExecute": false,
    "useLegacyReadLine": false,
    "forceClearScrollbackBuffer": false,
    "suppressStartupBanner": false
  },
  "sideBar": {
    "CommandExplorerVisibility": true,
    "CommandExplorerExcludeFilter": []
  },
  "pester": {
    "useLegacyCodeLens": true,
    "outputVerbosity": "FromPreference",
    "debugOutputVerbosity": "Diagnostic"
  },
  "buttons": {
    "showRunButtons": true,
    "showPanelMovementButtons": true
  },
  "cwd": "",
  "enableReferencesCodeLens": true,
  "analyzeOpenDocumentsOnly": false
}

5/18/2023 8:31:53 AM [DIAGNOSTIC] - Creating log directory at: 'file:///c%3A/Users/jodell/AppData/Roaming/Code/User/globalStorage/ms-vscode.powershell/logs/1684416713-a9fc9243-336d-40f5-a653-a94da5d0a2011684416708327'

5/18/2023 8:31:53 AM [NORMAL] - Visual Studio Code v1.78.2 64-bit

5/18/2023 8:31:53 AM [NORMAL] - PowerShell Extension v2023.5.0

5/18/2023 8:31:53 AM [NORMAL] - Operating System: Windows 64-bit

5/18/2023 8:31:53 AM [NORMAL] - Language server starting --

5/18/2023 8:31:53 AM [NORMAL] -     PowerShell executable: C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe

5/18/2023 8:31:53 AM [NORMAL] -     PowerShell args: -NoProfile -ExecutionPolicy Bypass -Command Import-Module 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2023.5.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules' -EnableConsoleRepl -StartupBanner "PowerShell Extension v2023.5.0
Copyright (c) Microsoft Corporation.

https://aka.ms/vscode-powershell
Type 'help' to get help.
" -LogLevel 'Diagnostic' -LogPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\logs\1684416713-a9fc9243-336d-40f5-a653-a94da5d0a2011684416708327\EditorServices.log' -SessionDetailsPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\sessions\PSES-VSCode-16148-872537.json' -FeatureFlags @() 

5/18/2023 8:31:53 AM [NORMAL] -     PowerShell Editor Services args: Import-Module 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2023.5.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\jodell\.vscode\extensions\ms-vscode.powershell-2023.5.0\modules' -EnableConsoleRepl -StartupBanner "PowerShell Extension v2023.5.0
Copyright (c) Microsoft Corporation.

https://aka.ms/vscode-powershell
Type 'help' to get help.
" -LogLevel 'Diagnostic' -LogPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\logs\1684416713-a9fc9243-336d-40f5-a653-a94da5d0a2011684416708327\EditorServices.log' -SessionDetailsPath 'c:\Users\jodell\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\sessions\PSES-VSCode-16148-872537.json' -FeatureFlags @() 

5/18/2023 8:31:54 AM [NORMAL] - powershell.exe started.

5/18/2023 8:33:50 AM [NORMAL] - powershell.exe PID: 18820

5/18/2023 8:33:50 AM [NORMAL] - Waiting for session file...

5/18/2023 8:34:20 AM [WARNING] - Loading the PowerShell extension is taking longer than expected. If you're using privilege enforcement software, this can affect start up performance.

**Comment by the-joey-o** [comment]
A couple more logs were written after a few more minutes:

5/18/2023 8:37:51 AM [NORMAL] - Timed out waiting for session file to appear!

5/18/2023 8:37:55 AM [ERROR] - PowerShell language server process didn't start!

**Comment by andschwa** [comment]
When you open the PowerShell Extension Logs folder, there should be a log file called `EditorServices.log` that has what I'm hoping to see, especially if you managed to set it the logging level to Diagnostic: `"powershell.developer.editorServicesLogLevel": "Diagnostic"`

**Comment by the-joey-o** [comment]
I have the PowerShell Editor Services Log Level set to Diagnostic, but the log is empty.
![image](https://github.com/PowerShell/vscode-powershell/assets/7991004/e25abcd7-9988-4384-94d2-5f9baa9e0531)
![image](https://github.com/PowerShell/vscode-powershell/assets/7991004/fedd473a-a861-4d2e-8351-be7b4df81fed)

**Comment by andschwa** [comment]
Hm, ok. Can you try PowerShell 7? That'll at least help us determine if it's specific to PowerShell 5.1 or something else.

**Comment by the-joey-o** [comment]
Switching to Version 7 seems to have fixed the problem.
![image](https://github.com/PowerShell/vscode-powershell/assets/7991004/9bc03ccd-344d-4905-a45b-b55ea8c589db)

**Comment by andschwa** [comment]
That's helpful, thank you! So I've spent the last week revamping our logging and start/stop/restart logic. When I get a new preview out I will have you test again, it'll probably repro still _but_ I'm hoping we'll have a lot more information available to track it down.

**Comment by andschwa** [comment]
@the-joey-o have you been able to try the latest pre-release? Is it working better for you now?

**Comment by aproposts** [comment]
I'm also unable to start the extension terminal with 5.1 and the latest VSCode.

After getting tired of waiting then hitting Ctrl-C:
![image](https://github.com/PowerShell/vscode-powershell/assets/110621029/3d2f2aa3-b3f4-4f21-9db6-70102a67caf5)

Here's the main error text:
`The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe '-NoProfile', '-ExecutionPolicy', 'Bypass', '-Command', 'Import-Module 'c:\Users\thompsonc\.vscode\extensions\ms-vscode.powershell-2023.6.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2023.6.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\thompsonc\.vscode\extensions\ms-vscode.powershell-2023.6.0\modules' -EnableConsoleRepl -StartupBanner '' -LogLevel 'Diagnostic' -LogPath 'c:\Users\thompsonc\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\logs\1687193178-2a3208d9-287b-4617-bdb7-6d51145aa4ab1687193157982\EditorServices.log' -SessionDetailsPath 'c:\Users\thompsonc\AppData\Roaming\Code\User\globalStorage\ms-vscode.powershell\sessions\PSES-VSCode-20908-838122.json' -FeatureFlags @() '" terminated with exit code: 2.`

I've attached a zip of the logs folder at the indicated location:
[vscode-powershell.zip](https://github.com/PowerShell/vscode-powershell/files/11790507/vscode-powershell.zip)

Notably, the referenced `SessionDetailsPath` doesn't have a file by that name at that location. Presumably, that's what we collect the Editor Services log for:
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/11790523/EditorServices.log)

For completeness:

`code --version`:

```text
1.79.2
 695af097c7bd098fbf017ce3ac85e09bbc5dda06
 x64
```

`code --list-extensions --show-versions | Select-String powershell`

`ms-vscode.powershell@2023.6.0`

The following are run after selecting 'PowerShell (x64)' from 'Show Session Menu':

```powershell
PS> $psEditor.EditorServicesVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
3      8      7      0       

PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by andschwa** [comment]
@aproposts it looks like it hung when loading profiles...can you share what's in your PowerShell profiles, and can you try to repro but with `powershell.enableProfileLoading` set to false?

**Comment by aproposts** [comment]
@andschwa, I just disabled profile loading, with seemingly the same result. I can capture the same set of logs if you like.

**Comment by aproposts** [comment]
Follow-up: My problem 'mysteriously' disappeared. I opened a folder this morning (which I have configured to use PS5 for the extension) expecting it to fail--and it just worked.

Since my last message, I updated Windows and rebooted. `code.exe` and extension versions are exactly the same. I opened the same folder from which I was working when I captured the logs: PS5, no problem. I'm a bit flummoxed as to why/how this is possible. It doesn't look like I changed anything in my settings either... And of course it's staring me right in the face: I modified my profile before committing, pushing and linking. I'll see if I can trace back through my changes to find the offending section.
